lighthouse-core/gather/gather-runner.js

```diff
@@ -301,6 +319,9 @@ class GatherRunner {
   static collectArtifacts(gathererResults) {
     const artifacts = {};
 
+    // Nest LighthouseRunWarnings, if any, so they will be collected into artifact.
+    gathererResults.LighthouseRunWarnings = [gathererResults.LighthouseRunWarnings || []];
```

Comment:
shouldn't `gathererResults.LighthouseRunWarnings` always be set? if not, do we need a safeguard in warnOnHeadless too

Reply:
> shouldn't gathererResults.LighthouseRunWarnings always be set?

it should always be set, I was thinking of `runner` when I added that :)